planner/optimize.go

```diff
@@ -38,12 +39,23 @@ import (
 // Optimize does optimization and creates a Plan.
 // The node must be prepared first.
 func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plannercore.Plan, types.NameSlice, error) {
-	fp := plannercore.TryFastPlan(sctx, node)
-	if fp != nil {
-		if !useMaxTS(sctx, fp) {
-			sctx.PrepareTSFuture(ctx)
+	_, isolationReadContainTiKV := sctx.GetSessionVars().GetIsolationReadEngines()[kv.TiKV]
+	if isolationReadContainTiKV {
+		cfgIsolationReadContainTiKV := false
+		for _, engine := range config.GetGlobalConfig().IsolationRead.Engines {
```

Comment:
why we need to check the config level variables since the session level variable has already indicate this session to read from tikv storage.

Reply:
That's defferent scope, config means instance scope and session is session scope. The intersection of them is the set of isolation engines.

Reply:
By default, the value of session level variables should be equal to the config values. Once users changed the session variables, we should only check the session variables.

That is, session level values can be regraded as local variables, whereas config level values can be regarded as global variables in programming language.

This should be the behavior of session <- server <- global scope variables.

Reply:
Yes, that's what we should do after delete the global scope. Not in this PR.